be/src/exec/pipeline/sort/partition_sort_sink_operator.cpp

```diff
@@ -52,6 +58,24 @@ Status PartitionSortSinkOperator::push_chunk(RuntimeState* state, const ChunkPtr
                                                                          _sort_exec_exprs, _order_by_types);
     RETURN_IF_ERROR(materialize_chunk);
     TRY_CATCH_BAD_ALLOC(RETURN_IF_ERROR(_chunks_sorter->update(state, materialize_chunk.value())));
+
+    const auto& build_runtime_filters = _sort_context->build_runtime_filters();
+    if (!build_runtime_filters.empty()) {
+        auto runtime_filter = _chunks_sorter->runtime_filters(state->obj_pool());
+        if (runtime_filter == nullptr) {
+            return Status::OK();
+        }
+        DCHECK_EQ(runtime_filter->size(), build_runtime_filters.size());
+        std::list<RuntimeFilterBuildDescriptor*> build_descs(build_runtime_filters.begin(),
+                                                             build_runtime_filters.end());
+        for (size_t i = 0; i < build_runtime_filters.size(); ++i) {
+            build_runtime_filters[i]->set_or_intersect_filter((*runtime_filter)[i]);
+            RuntimeBloomFilterList lst = {build_runtime_filters[i]};
+            _hub->set_collector(_plan_node_id,
+                                std::make_unique<RuntimeFilterCollector>(RuntimeInFilterList{}, std::move(lst)));
+        }
+        state->runtime_filter_port()->publish_runtime_filters(build_descs);
```

Comment:
Here will publish multi times? or just publish once is enogh?

Reply:
publish local filter multi times is ok. In fact only the first time works. publish multi times will be ignored